src/IssueViz/Models/AnalysisIssueLocationVisualization.cs

```diff
@@ -85,6 +85,11 @@ public string CurrentFilePath
             {
                 filePath = value;
                 NotifyPropertyChanged();
+
+                if (string.IsNullOrEmpty(filePath))
+                {
+                    this.InvalidateSpan();
```

Comment:
`InvalidateSpan` is also going to raise a change notification. Wouldn't it be better to make all the state changes first before any of the notifications are raised so that the properties are in a consistent state when any listeners are notified?

i.e. move the notification in line 87 to after this new block of code, so the sequence of operations is
1. set filePath
2. set span
3. notify span
4. notify filePath